src/sys/wait.rs

```diff
@@ -215,24 +262,84 @@ fn decode(pid : Pid, status: i32) -> WaitStatus {
     }
 }
 
-pub fn waitpid<P: Into<Option<Pid>>>(pid: P, options: Option<WaitPidFlag>) -> Result<WaitStatus> {
+/// Designates whether the supplied `Pid` value is a process ID, process group ID,
+/// specifies any child of the current process's group ID, or any child of the current process.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum PidGroup {
+    /// Signifies that the `Pid` is supposed to represent a process ID.
+    ProcessID(Pid),
+    /// Signifies that the `Pid` is supposed to represent a process group ID.
+    ProcessGroupID(Pid),
+    /// Signifies that any child process that belongs to the process group of the current process
+    /// should be selected.
+    AnyGroupChild,
+    /// Signifies that any child process that belongs to the current process should be selected.
+    AnyChild,
+}
+
+impl From<i32> for PidGroup {
+    fn from(pid: i32) -> PidGroup {
+        if pid > 0 {
+            PidGroup::ProcessID(Pid::from_raw(pid))
+        } else if pid < -1 {
+            PidGroup::ProcessGroupID(Pid::from_raw(pid))
+        } else if pid == 0 {
+            PidGroup::AnyGroupChild
+        } else {
+            PidGroup::AnyChild
+        }
+    }
+}
+
+/// Waits for and returns events that are received from the given supplied process or process group
```

Comment:
This should be a short descriptive line, then a blank line, then more description. If you run `rust doc --no-deps --open` and look at the generated output, you'll see why. The first line is used on the summary page, and the rest is shown on the detailed page.

Additionally this should link to the [man page](http://pubs.opengroup.org/onlinepubs/007908799/xsh/wait.html) (see how other functions are documented).